tests/test_browsers_playlists.py

```diff
@@ -337,25 +347,27 @@ def test_playlist_deletion_CANCEL(self):
         self.failUnlessEqual(b.playlists()[1], second_pl)
 
     def test_import(self):
-        pl_name = u"_€3 œufs à Noël"
-        pl = FileBackedPlaylist(_TEMP_DIR, pl_name, None)
-        pl.extend(SONGS)
-        pl.write()
-        new_fn = os.path.splitext(text2fsn(pl.name))[0] + '.m3u'
-        new_path = os.path.join(pl.dir, new_fn)
-        os.rename(pl.path, new_path)
-        added = self.bar._import_playlists([new_path], self.lib)
-        self.failUnlessEqual(added, 1, msg="Failed to add '%s'" % new_path)
-        os.unlink(new_path)
+        def fns(songs):
+            return [song('~filename') for song in songs]
+        pl_lib = self.bar.pl_lib
+        assert len(self.bar.playlists()) == 2, "Should start with two playlists"
+        assert len(pl_lib) == 2, f"Started with {pl_lib.keys()}"
+
+        pl_name = "_€3 œufs à Noël"
+        pl_path = Path(_TEMP_DIR) / (pl_name + ".m3u")
+        with open(pl_path, "wb") as f:
+            for fn in fns(SONGS):
+                f.write(fsn2bytes(fn, "utf-8") + b"\n")
+        pls_added, songs_added = self.bar._import_playlists([str(pl_path)])
+        assert pls_added == 1, f"Failed to add {pl_path}"
+        assert len(self.bar.songs_lib) == len(self.ALL_SONGS)
+        assert songs_added == 0, "Why did we add existing songs?"
+        assert len(pl_lib) == 3, f"Got PLs: \n{', '.join(str(pl) for pl in pl_lib)}"
         pls = self.bar.playlists()
-        self.failUnlessEqual(len(pls), 3)
+        assert len(pls) == 3, f"Got PL rows: {', '.join(str(pl) for pl in pls)}"
         # Leading underscore makes it always the last entry
         imported = pls[-1]
-        self.failUnlessEqual(imported.name, pl_name)
-
-        def fns(songs):
-            return [song('~filename') for song in songs]
-        self.failUnlessEqual(fns(imported.songs), fns(pl.songs))
+        self.failUnlessEqual(fns(imported.songs), fns(SONGS))
```

Comment:
Is there a benefit to using `assert` in some cases, but `failUnlessEqual` in others, or vice versa?

Reply:
Using native `assert` in tests is a nice [feature of pytest](https://docs.pytest.org/en/stable/assert.html#asserting-with-the-assert-statement) and IMO looks much neater. Unittest is kinda ugly and weird now.

But, nobody fancies changing the entire codebase over so I've just started doing it for new / modified things, when I remember...

Reply:
> Using native assert in tests is a nice feature of pytest and IMO looks much neater. Unittest is kinda ugly and weird now.

Agreed (on both). There's also a `self.assertTrue` and `assertFalse` :sweat_smile: 

> But, nobody fancies changing the entire codebase over so I've just started doing it for new / modified things, when I remember...

True. I think it might be possible to automate this. I might try this..

```python
assertTrue(x, msg) -> assert x, msg 

# now ignore msg parameter
assertTrue(x) -> assert x 
assertFalse(x) -> assert not x
assertEquals(x, y) -> assert x == y
assertNotEquals(x, y) -> assert x != y
assertEqual(x, y) -> assert x == y
assertNotEqual(x, y) -> assert x != y
failUnlessEqual(x, y) -> assert x == y
failIfEqual(x, y) -> assert x != y
failUnless(x) -> assert x
failIf(x) -> assert not x
failUnlessAlmostEqual -> assertAlmostEqual
failIfAlmostEqual -> assertNotAlmostEqual
failUnlessRaises -> assertRaises
```

Reply:
Sounds good though it would need human review a bit. 

Yes the message bit gets hard especially with PEP-8 constraints. Also beware that pytest is a bit more opinionated / normal than some QL's tests in ordering of `assert actual == expected` etc, so a few things may need swapping